src/main/java/io/aiven/kafka/connect/transforms/ExtractTimestamp.java

```diff
@@ -70,8 +73,13 @@ public R apply(final R record) {
         }
 
         final long newTimestamp;
-        if (fieldValue instanceof Long) {
-            newTimestamp = (long) fieldValue;
+        if (fieldValue instanceof Long) {            
+            if ((long) fieldValue < 10000000000L) {
+                newTimestamp = (long) fieldValue * 1000L;
+            } else {
+                newTimestamp = (long) fieldValue;
+            }
```

Comment:
AFAIU, the intention of this is to support seconds as well as milliseconds. However, doing this implicitly is a bad idea as there might be cases where this is not what the user want.

If we have a request to support seconds, this needs configuration, documentation, and tests. Otherwise, I'd drop this for now.